### PR TITLE
Expose KeyedRateLimiter.len()

### PR DESCRIPTION
**Pull request by lytefast**
This will enable a few things:
- stats tracking
- some strategies around when to do maintenance on the map

### CODE REVIEW

**Review by antifuchs — APPROVED** [review]
Ah! This is indeed a thing I missed. Thank you!

I'll port this over to [`governor`](https://github.com/antifuchs/governor) as well.

**Comment by antifuchs** [comment]
Before this can go in, master's clippy must become happy again; I'll action this now, as well.

File: src/state/keyed.rs [hunk 1]
```diff
@@ -134,6 +134,11 @@ where
         }
     }
 
+    /// Returns the number of non-empty keys present in the map.
+    pub fn len(&self) -> usize {
+        self.map_reader.len()
+    }
```
Comment (antifuchs) [inline src/state/keyed.rs:140]: Oh, wait - turns out the `clippy` failure wasn't from the master branch at all:

> error: item `state::keyed::KeyedRateLimiter<K, A, C>` has a public `len` method but no corresponding `is_empty` method

Can you implement that one as well?

**Review by antifuchs — CHANGES_REQUESTED** [review]
Eep, missed that clippy complains about changes on this branch!

File: src/state/keyed.rs (hunk 1 above)
Comment (lytefast) [reply src/state/keyed.rs:140]: Yup!

**Comment by lytefast** [comment]
Side, note. I didn't know that https://github.com/antifuchs/governor was the successor to this library, might be worth noting in the `README.md`.

**Review by antifuchs — APPROVED** [review]

**Pull request merged**